lib/msf/core/rpc/v10/rpc_session.rb

```diff
@@ -5,6 +5,16 @@ module Msf
 module RPC
 class RPC_Session < RPC_Base
 
+  def initialize(service)
+    @deprecator = ActiveSupport::Deprecation.new('7.0', 'metasploit-framework')
```

Comment:
I'm open to discussion about removal version for deprecated RPC methods. Feels like "Next major version" makes the most sense